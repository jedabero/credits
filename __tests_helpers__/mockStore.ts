// eslint-disable-next-line import/no-extraneous-dependencies
import configureMockStore from 'redux-mock-store';
import { RootState } from '../src/store';

export default configureMockStore<RootState>();
