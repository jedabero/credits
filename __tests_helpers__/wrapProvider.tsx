import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { Store } from '../src/store';

const wrapProvider = (component: ReactElement, store: Store) => (
  <Provider store={store}>{component}</Provider>
);

export default wrapProvider;
