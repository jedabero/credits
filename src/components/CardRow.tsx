import { useCallback } from 'react';
import { EntityId } from '@reduxjs/toolkit';

import { selectCreditCardById, removeCard } from '../reducers/creditCards';
import { useAppSelector, useAppDispatch } from '../hooks';

type CardRowProps = {
  cardId: EntityId,
  onEdit: React.Dispatch<React.SetStateAction<EntityId | undefined>>,
  onView: React.Dispatch<React.SetStateAction<EntityId | undefined>>,
};

function CardRow({ cardId, onEdit, onView }: CardRowProps) {
  const dispatch = useAppDispatch();
  const handleDelete = useCallback(() => dispatch(removeCard(cardId)), [cardId, dispatch]);
  const handleEdit = useCallback(() => onEdit(cardId), [cardId, onEdit]);
  const handleView = useCallback(() => onView(cardId), [cardId, onView]);
  const card = useAppSelector((state) => selectCreditCardById(state, cardId));
  if (!card) return null;
  return (
    <div className="cardrow">
      <div>{(`${card.id}`).substring(0, 13)}</div>
      <div>{card.name}</div>
      <div>{card.cardNumber}</div>
      <div>{card.quota.toLocaleString('en-419', { style: 'currency', currency: 'USD' })}</div>
      <div>{card.balance.toLocaleString('en-419', { style: 'currency', currency: 'USD' })}</div>
      <div>{card.cutoffDay}</div>
      <div>
        <button type="button" onClick={handleView}>View</button>
        <button type="button" onClick={handleEdit}>Edit</button>
        <button type="button" onClick={handleDelete}>Delete</button>
      </div>
    </div>
  );
}

export default CardRow;
