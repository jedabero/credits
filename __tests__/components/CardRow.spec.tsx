/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { create } from 'react-test-renderer';
import CardRow from '../../src/components/CardRow';
import wrapProvider from '../../__tests_helpers__/wrapProvider';
import mockStore from '../../__tests_helpers__/mockStore';

describe('CardRow test', () => {
  test('should render', () => {
    const store = mockStore({
      _persist: { version: 0, rehydrated: true },
      creditCards: {
        entities: {},
        ids: [],
      },
      movements: {
        entities: {},
        ids: [],
      },
    });
    const tree = create(
      wrapProvider(
        <CardRow cardId="SOME ID" onEdit={() => {}} onView={() => {}} />,
        store,
      ),
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
