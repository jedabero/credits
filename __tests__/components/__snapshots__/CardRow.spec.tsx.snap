// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardRow test should render 1`] = `null`;
